lib/chef/resource/windows_firewall_profile.rb

```diff
@@ -0,0 +1,168 @@
+      property :default_inbound_block, [true, false, nil], default: true, description: "Set the default policy for inbound network traffic to blocked"
+      property :default_outbound_allow, [true, false, nil], default: true, description: "Set the default policy for outbound network traffic to allowed"
+      property :allow_inbound_rules, [true, false, nil], description: "Allow users to set inbound firewall rules"
+      property :allow_local_firewall_rules, [true, false, nil], description: "Merges inbound firewall rules into the policy"
+      property :allow_local_ipsec_rules, [true, false, nil], description: "Allow users to manage local connection security rules"
+      property :allow_user_apps, [true, false, nil], description: "Allow user applications to manage firewall"
+      property :allow_user_ports, [true, false, nil], description: "Allow users to manage firewall port rules"
+      property :allow_unicast_response, [true, false, nil], description: "Allow unicast responses to multicast and broadcast messages"
+      property :display_notification, [true, false, nil], description: "Display a notification when firewall blocks certain activity"
+
+      load_current_value do |desired|
+        ps_results = powershell_exec(<<-CODE).result
+          $#{desired.profile} = Get-NetFirewallProfile -Profile #{desired.profile}
+          $#{desired.profile}.Enabled
+          $#{desired.profile}.DefaultInboundAction
+          $#{desired.profile}.DefaultOutboundAction
+          $#{desired.profile}.AllowInboundRules
+          $#{desired.profile}.AllowLocalFirewallRules
+          $#{desired.profile}.AllowLocalIPsecRules
+          $#{desired.profile}.AllowUserApps
+          $#{desired.profile}.AllowUserPorts
+          $#{desired.profile}.AllowUnicastResponseToMulticast
+          $#{desired.profile}.NotifyOnListen
+        CODE
+        profile_enabled case ps_results[0]; when 0 then false; when 1 then true; end
+        default_inbound_block case ps_results[1]; when 4 then true; when 2 then false; when 0 then nil; end
+        default_outbound_allow case ps_results[2]; when 4 then false; when 2 then true; when 0 then nil; end
+        allow_inbound_rules case ps_results[3]; when 0 then false; when 1 then true; when 2 then nil; end
+        allow_local_firewall_rules case ps_results[4]; when 0 then false; when 1 then true; when 2 then nil; end
+        allow_local_ipsec_rules case ps_results[5]; when 0 then false; when 1 then true; when 2 then nil; end
+        allow_user_apps case ps_results[6]; when 0 then false; when 1 then true; when 2 then nil; end
+        allow_user_ports case ps_results[7]; when 0 then false; when 1 then true; when 2 then nil; end
+        allow_unicast_response case ps_results[8]; when 0 then false; when 1 then true; when 2 then nil; end
+        display_notification case ps_results[9]; when 0 then false; when 1 then true; when 2 then nil; end
+      end
+
+      action :configure do
```

Comment:
I'd change this action to :enable. That way you remove the need for the profile_enabled property. That just becomes the action and then you add a disable action that only does the converge_by disable. The